eclair-core/src/main/scala/fr/acinq/eclair/router/Announcements.scala

```diff
@@ -72,7 +72,13 @@ object Announcements {
 
   def makeNodeAnnouncement(nodeSecret: PrivateKey, alias: String, color: Color, nodeAddresses: List[NodeAddress], features: Features, timestamp: Long = System.currentTimeMillis.milliseconds.toSeconds): NodeAnnouncement = {
     require(alias.length <= 32)
-    val witness = nodeAnnouncementWitnessEncode(timestamp, nodeSecret.publicKey, color, alias, features, nodeAddresses, unknownFields = ByteVector.empty)
+    val sortedAddresses = nodeAddresses.map {
+      case address@(_: IPv4) => (1, address)
+      case address@(_: IPv6) => (2, address)
+      case address@(_: Tor2) => (3, address)
+      case address@(_: Tor3) => (4, address)
+    }.sortBy(_._1).map(_._2)
```

Comment:
We could have defined an `Ordering[NodeAddress]`, but that would probably be over engineering.

Reply:
I toyed with it :smiley: but decided that it was indeed a bit over-engineered